Rebuild using templates instead of execs

tedivm opened the pull request:
Previously this module did the majority of it's work using "execs", however it violated a lot of puppet best practices by not making those execs idempotent. Additionally it manually managed the service in a lot of ways rather than relying on the puppet providers to do so.

This update fixes a lot of that-

* There is now one `service.pp` file which should work on *all* systems.

* The configuration file itself is now built using puppet templates. This removes a huge number of the `exec` statements while also better integrating into the `Service` for logdna. If a change to the config file occurs the server gets issued the restart command (`logdna-agent restart`), but is no changes occur it is simply left alone.

* Parameter values have been changes slightly to meet puppet defaults. Specifically the `agent_service` parameter now only takes the same values as the `Service` `ensure` parameter so it can be passed directly, and numerous other parameters have been turned into empty arrays by default.

* Proper dependencies have been added, particularly in the installation section. This will ensure that packages aren't installed until after their repositories have been setup.

tedivm committed:
Rebuild using templates instead of execs

smusali commented:
@tedivm, most things look good but it seems you made a mistake in `configure.pp`; therefore, it's failing tests.

tedivm committed:
Remove tests for classes that don't exist, fix configure test

tedivm committed:
Remove configuration test (see commit comment)

The configuration class can not be compiled by itself- it depends on
the Service class to define the `logdna-agent` service so that when the
configuration changes it can restart the service.

This class *is* still tested, as it is called by the `logdna` class and
thus falls under that test.

tedivm commented:
@smusali - I didn't actually make a mistake, it's the test itself that is inaccurate. The test being used only works for completely standalone classes, but the configure class itself has a dependency on the Service class (specifically, the `Service['logdna-agent']` resource- this dependency is what forces the service to reload when the configuration changes).

The `logdna` test- which calls the `init` class- covers the `Configuration` class already, and does so with all of it's dependencies compiled. Since that test passes, and since the `Configuration` class by design is not an independent class, I am removing that test.

I also went ahead and merged the two different "service" tests down into one, to reflect the fact that there is only one service class now.

smusali committed:
Update configure.pp

tedivm commented on manifests/agent/configure.pp:
```diff
@@ -31,7 +31,7 @@
     Optional[String] $hostname             = $logdna::params::conf_hostname,
     Optional[Array[String]] $exclude       = $logdna::params::conf_exclude,
-    Optional[Array[String]] $exclude_regex = $logdna::params::conf_exclude_regex
+    Optional[String] $exclude_regex        = $logdna::params::conf_exclude_regex
```
This should still be an array. The template file will convert it back down to a single line string.

smusali replied:
I returned it back to being an array after merging but in `logdna-agent`, we take only one string value for the regex for exclusion.

Thanks again for this Pull Request!

smusali commented:
@tedivm, thanks for such a great work you have done - I have just changed type of `exclude_regex` from `Array[String]` to `String` since we read only one value in the agent.

Just published [v0.3.0](https://forge.puppet.com/logdna/logdna).
